scipy/optimize/tests/test_minpack.py

```diff
@@ -517,6 +517,26 @@ def test_maxfev_and_bounds(self):
         assert_allclose(popt1, 2, atol=1e-14)
         assert_allclose(popt2, 2, atol=1e-14)
 
+    def test_curvefit_covariance(self):
```

Comment:
I am not sure I understand the necessity, since the matrix used in solve_triangular is coming from cholesky, it will be lower triangular by definition. Am I wrong about this?